platform/core/src/classes/MetadataProvider.js

```diff
@@ -126,6 +147,52 @@ class MetadataProvider {
   _getCornerstoneDICOMImageLoaderTag(wadoImageLoaderTag, instance) {
     let metadata;
 
+    // NOTE : we only check if the flag is set and set to false to prevent other
+    // non compatible data source to work even if automaticallyLoadDicomMetadata
+    // is set.
+    // To make your data source compatible with this you'll have to explicitly
+    // set the instance.imageMetadataLoaded to false
+    if (this.automaticallyLoadDicomMetadata && (instance.imageMetadataLoaded === false)) {
```

Comment:
as I said above, I prefer not to call the dicomImageLoader here, but even with calling it here, I don't think we need a new flag. We can derive the flag status from whether it has all necessary metadata (we should create a schema for that somewhere to validate againts, that also helps people to know what metadata is necessary) and if not we automatically switch to the dicomloader+dcmjs how about that?

Reply:
@wayfarer3130 any thought on where to call the dicomImageLoader + dcmjs other than here? (ideally they should not be here)